### PR TITLE
[Identity] Add docstrings for WAM/MSA params

### PR DESCRIPTION
**Pull request by pvaneck**

### CODE REVIEW

File: sdk/identity/azure-identity/azure/identity/_credentials/browser.py [hunk 1]
```diff
@@ -43,6 +43,15 @@ class InteractiveBrowserCredential(InteractiveCredential):
         will cache tokens in memory.
     :paramtype cache_persistence_options: ~azure.identity.TokenCachePersistenceOptions
     :keyword int timeout: seconds to wait for the user to complete authentication. Defaults to 300 (5 minutes).
+    :keyword bool allow_broker: Brokers provide single sign-on, device identification, and application identification
```
Comment (joshfree) [inline sdk/identity/azure-identity/azure/identity/_credentials/browser.py:46]: I think this should read `An authentication broker is an application that runs on a user's machine that manages the authentication handshakes and token maintenance for connected accounts.` ....

File: sdk/identity/azure-identity/azure/identity/_credentials/device_code.py [hunk 2]
```diff
@@ -41,12 +42,17 @@ class DeviceCodeCredential(InteractiveCredential):
...
         If this argument isn't provided, the credential will print instructions to stdout.
     :paramtype prompt_callback: Callable[str, str, ~datetime.datetime]
+
     :keyword AuthenticationRecord authentication_record: :class:`AuthenticationRecord` returned by :func:`authenticate`
     :keyword bool disable_automatic_authentication: if True, :func:`get_token` will raise
         :class:`AuthenticationRequiredError` when user interaction is required to acquire a token. Defaults to False.
     :keyword cache_persistence_options: configuration for persistent token caching. If unspecified, the credential
         will cache tokens in memory.
     :paramtype cache_persistence_options: ~azure.identity.TokenCachePersistenceOptions
+    :keyword bool allow_broker: Brokers provide single sign-on, device identification, and application identification
```
Comment (joshfree) [inline sdk/identity/azure-identity/azure/identity/_credentials/device_code.py:52]: same feedback on phrasing here. See https://learn.microsoft.com/en-us/entra/msal/dotnet/acquiring-tokens/desktop-mobile/wam#what-is-a-broker

**Review by joshfree — APPROVED** [review]

File: sdk/identity/azure-identity/azure/identity/_credentials/device_code.py (hunk 2 above)
Comment (pvaneck) [reply sdk/identity/azure-identity/azure/identity/_credentials/device_code.py:52]: Thanks for the pointer! WIll update.

File: sdk/identity/azure-identity/azure/identity/_credentials/device_code.py [hunk 3]
```diff
@@ -41,12 +42,19 @@ class DeviceCodeCredential(InteractiveCredential):
...
         If this argument isn't provided, the credential will print instructions to stdout.
     :paramtype prompt_callback: Callable[str, str, ~datetime.datetime]
+
     :keyword AuthenticationRecord authentication_record: :class:`AuthenticationRecord` returned by :func:`authenticate`
     :keyword bool disable_automatic_authentication: if True, :func:`get_token` will raise
         :class:`AuthenticationRequiredError` when user interaction is required to acquire a token. Defaults to False.
     :keyword cache_persistence_options: configuration for persistent token caching. If unspecified, the credential
         will cache tokens in memory.
     :paramtype cache_persistence_options: ~azure.identity.TokenCachePersistenceOptions
+    :keyword bool allow_broker: An authentication broker is an application that runs on a user's machine that manages
```
Comment (xiangyan99) [inline sdk/identity/azure-identity/azure/identity/_credentials/device_code.py:52]: WAM does not work for device code.

We only need browser and username/password.

**Comment by pvaneck** [comment]
Closed in favor of https://github.com/Azure/azure-sdk-for-python/pull/32360

**Pull request closed without merging**